packages/financial-templates-lib/src/price-feed/UniswapPriceFeed.js

```diff
@@ -86,14 +82,40 @@ class UniswapPriceFeed extends PriceFeedInterface {
 
   // Not part of the price feed interface. Can be used to pull the uniswap price at the most recent block.
   getLastBlockPrice() {
-    return this.lastBlockPrice && this.convertPoolDecimalsToPriceFeedDecimals(this.lastBlockPrice);
+    return this.lastBlockPrice && this.convertBaseDecimalsToPriceFeedDecimals(this.lastBlockPrice);
   }
 
   getPriceFeedDecimals() {
     return this.priceFeedDecimals;
   }
 
   async update() {
+    // Read token0 and token1 precision from Uniswap contract.
+    const [token0Address, token1Address] = await Promise.all([
+      this.uniswap.methods.token0().call(),
+      this.uniswap.methods.token1().call()
+    ]);
+    this.token0 = new this.web3.eth.Contract(this.erc20Abi, token0Address);
+    this.token1 = new this.web3.eth.Contract(this.erc20Abi, token1Address);
+    const [token0Precision, token1Precision] = await Promise.all([
+      this.token0.methods.decimals().call(),
```

Comment:
what happens if the token does not have a decimals method?

Reply:
I think we can handle this two ways. One ignore it and assume that we can only work with pools that have tokens that implement decimals. I think this is reasonable since any tokens that don't implement decimals might not be that robust. [This is the same assumption that the `liquidator/index.js` makes for example](https://github.com/UMAprotocol/protocol/blob/master/packages/liquidator/index.js#L159).

Or we can catch and assume 18 decimals.

Reply:
I think either is probably fine. I'm not sure of many examples of tokens that don't implement decimals, but I would assume that any that don't are probably 18. It's probably a _little_ safer to throw an error in the absence of the method, however.

Reply:
Yeah agreed I'd like to know if a token doesn't implement `decimals()`